### PR TITLE
Added tests to ensure copied references are consistent

### PR DESCRIPTION
**Pull request by dantleech**
Ensure that references within a copied set remain consistent.

i.e. if a node with a set references another node within the set, then the reference in the new set should not reference the old set.

### CODE REVIEW

**Comment by dbu** [comment]
this leads to a failing test with jackalope-jackrabbit. can you investigate why this happens?

```
1) PHPCR\Tests\Writing\CopyMethodsTest::testWorkspaceCopyReference
PHPCR\NodeType\ConstraintViolationException: HTTP 409: no matching child node definition found for {}copiedReferencedSet

/home/david/liip/jackalope/jackalope-jackrabbit/src/Jackalope/Transport/Jackrabbit/Request.php:544
/home/david/liip/jackalope/jackalope-jackrabbit/src/Jackalope/Transport/Jackrabbit/Request.php:483
/home/david/liip/jackalope/jackalope-jackrabbit/src/Jackalope/Transport/Jackrabbit/Request.php:374
/home/david/liip/jackalope/jackalope-jackrabbit/src/Jackalope/Transport/Jackrabbit/Client.php:1056
/home/david/liip/jackalope/jackalope-jackrabbit/src/Jackalope/Transport/Jackrabbit/Client.php:1037
/home/david/liip/jackalope/jackalope-jackrabbit/vendor/jackalope/jackalope/src/Jackalope/Transport/AbstractReadWriteLoggingWrapper.php:31
/home/david/liip/jackalope/jackalope-jackrabbit/vendor/jackalope/jackalope/src/Jackalope/ObjectManager.php:1443
/home/david/liip/jackalope/jackalope-jackrabbit/vendor/jackalope/jackalope/src/Jackalope/Workspace.php:112
/home/david/liip/jackalope/jackalope-jackrabbit/vendor/phpcr/phpcr-api-tests/tests/10_Writing/CopyMethodsTest.php:79
```

File: fixtures/10_Writing/copy.xml [hunk 1]
```diff
@@ -1,348 +1,424 @@
...
+                    </sv:property>
+                    <sv:property sv:name="someProperty" sv:type="String">
+                        <sv:value>yo</sv:value>
+                    </sv:property>
+                </sv:node>
+            </sv:node>
+        </sv:node>
+        <sv:node sv:name="dstNode">
+            <sv:property sv:name="jcr:primaryType" sv:type="Name">
+                <sv:value>nt:folder</sv:value>
```
Comment (dbu) [inline fixtures/10_Writing/copy.xml]: if the destination is a folder, you can only move things in here that are nt:hiearchyNode - it seems jackalope-doctrine-dbal is missing some validation during move operations.

**Comment by dbu** [comment]
ok, found it. i suggest you make dstNode nt:unstructured too.

**Comment by dantleech** [comment]
ok, updated..

**Comment by dantleech** [comment]
(and tested with jackrabbit)

**Comment by dantleech** [comment]
btw. I also has to fix the fixtures in resepect to the naming of the properties and child nodes.

the `src` node has both the property `block_1` and child node `block_1` - jackrabbit doesn't like that at all and the property was not found (only the child node).

This works in doctrine-dbal however - which seems odd as they both use the same `stdClass` hack.

Just for the record - is it legal in PHPCRfor child nodes and properties to share the same names?

**Comment by dbu** [comment]
afaik it does not work when reading, the property is hidden by the child. basically jackrabbit uses invalid json to transfer that info and its own json parser can handle the situation.

with stdclass i would not expect this to work at all.

**Comment by dbu** [comment]
thanks!

**Pull request merged**